pkg/handlers/proxy_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package handlers
+
+import (
+	"net/http"
+	"testing"
+)
+
+type headerTest struct {
+	key      string // header key
+	val      string // header val
+	expected string // expected result
+}
+
+func TestGetScheme(t *testing.T) {
+	headers := []headerTest{
+		{xForwardedProto, "https", "https"},
+		{xForwardedProto, "http", "http"},
+		{xForwardedProto, "HTTP", "http"},
+		{xForwardedScheme, "https", "https"},
+		{xForwardedScheme, "http", "http"},
+		{xForwardedScheme, "HTTP", "http"},
+		{forwarded, `For="[2001:db8:cafe::17]:4711`, ""},                    // No proto
+		{forwarded, `for=192.0.2.43, for=198.51.100.17;proto=https`, ""},    // Multiple params, will be empty.
+		{forwarded, `for=172.32.10.15; proto=https;by=127.0.0.1;`, "https"}, // Space before proto
+		{forwarded, `for=192.0.2.60;proto=http;by=203.0.113.43`, "http"},    // Multiple params
```

Comment:
for completeness, you could the following test case,
```go
headerTest{
  key : forwarded, 
  val: `for=client-ip,for=proxy1;proto=https,for=proxy2;proto=http`,
  expected:  "",
}
```
This tests validates that `GetSourceScheme` gets scheme from the first entry, which represents the client.

Reply:
i have already added that for=192.0.2.43, for=198.51.100.17;proto=https @krisis i don't see a value in adding one more though.

Reply:
@harshavardhana oops, I missed it.